File: result/translate/transcoder/java2python/python/PRIMALITY_TEST_SET_5USING_LUCAS_LEHMER_SERIES.py
def f_gold ( p ) :
    check_number = long ( pow ( 2 , p ) ) - 1
    nextval = 4 % check_number
    for i in range ( 1 , p - 1 ) :
        nextval = ( nextval * nextval - 2 ) % check_number
    return ( nextval == 0 )

File: result/translate/starchat/java2cpp/c++/PRIMALITY_TEST_SET_5USING_LUCAS_LEHMER_SERIES.cpp
bool f_gold ( int p ) {
    long checkNumber = pow ( 2, p ) - 1;
    long nextval = 4 % checkNumber;
    for ( int i = 1; i < p - 1; i++ ) nextval = ( nextval * nextval - 2 ) % checkNumber;
    return ( nextval == 0 );
}